bin/peggy.mjs

```diff
@@ -0,0 +1,331 @@
+
+// Helpers
+
+function abort(message) {
+  console.error(message);
+  process.exit(1);
+}
+
+function abortError(msg, error) {
+  console.error(msg);
+  if (typeof error.format === "function") {
+    abort(error.format([{
+      source: testGrammarSource,
+      text: testText,
+    }]));
+  } else {
+    if (verbose) {
+      abort(error);
+    } else {
+      abort(`Error: ${error.message}`);
+    }
+  }
+}
+
+function addExtraOptions(options, json) {
+  let extraOptions;
+
+  try {
+    extraOptions = JSON.parse(json);
+  } catch (e) {
+    if (!(e instanceof SyntaxError)) { throw e; }
+
+    abortError("Error parsing JSON:", e);
+  }
+  if ((extraOptions === null)
+      || (typeof extraOptions !== "object")
+      || Array.isArray(extraOptions)) {
+    abort("The JSON with extra options has to represent an object.");
+  }
+  return Object.assign({}, options, extraOptions);
+}
+
+// Files
+
+function readStream(inputStream, callback) {
+  let input = "";
+  inputStream.on("data", data => { input += data; });
+  inputStream.on("end", () => { callback(input); });
+}
+
+function readFile(name) {
+  let f = null;
+  try {
+    f = fs.readFileSync(name, "utf8");
+  } catch (e) {
+    abortError(`Can't read from file "${name}".`, e);
+  }
+  return f;
+}
+
+// Command line processing
+
+const program = new Command();
+let options = program
+  .version(peg.VERSION, "-v, --version")
+  .arguments("[input_file]")
+  .addOption(
+    new Option(
+      "--allowed-start-rules <rules>",
+      "comma-separated list of rules the generated parser will be allowed to start parsing from"
+    )
+      .default([], "the first rule in the grammar")
+      .argParser((val, prev) => prev.concat(val.split(",").map(x => x.trim())))
+  )
+  .option(
+    "--cache",
+    "make generated parser cache results"
+  )
+  .option(
+    "-d, --dependency <dependency>",
+    "use specified dependency (can be specified multiple times)",
+    (val, prev) => (prev || []).concat([val])
+  )
+  .option(
+    "-e, --export-var <variable>",
+    "name of a global variable into which the parser object is assigned to when no module loader is detected"
+  )
+  .option(
+    "--extra-options <options>",
+    "additional options (in JSON format as an object) to pass to peg.generate",
+    (val, prev) => addExtraOptions(prev, val)
+  )
+  .option(
+    "-c, --extra-options-file <file>",
+    "file with additional options (in JSON or commonjs module format as an object) to pass to peg.generate",
+    (val, prev) => {
+      if (/\.c?js$/.test(val)) {
+        return Object.assign({}, prev, require(path.resolve(val)));
+      } else {
+        return addExtraOptions(prev, readFile(val));
+      }
+    }
+  )
+  .addOption(
+    new Option(
+      "--format <format>",
+      "format of the generated parser"
+    )
+      .choices(MODULE_FORMATS)
+      .default(DEFAULT_FORMAT, '"commonjs"')
+  )
+  .option("-o, --output <file>", "output file")
+  .option(
+    "--plugin <module>",
```

Comment:
I don't think this is as clean as just doing `--plugin foo --plugin bar`, because otherwise you'll often need to use `--` to stop the list of plugins.

Reply:
According to examples you will need the stop marker only if plugins were the last option in the arguments list. If you have additional flags after it the stopper will no required, i.e. you can write:
```sh
peggy --plugin a b c -с options.json grammar.peggy
```
and that will work. However this is still a matter of personal preferences, so you can leave it as is